pennylane/pauli/pauli_arithmetic.py

```diff
@@ -414,10 +440,61 @@ def _pw_wires(w: Iterable) -> wires.Wires:
 
         return math.expand_matrix(reduced_mat, result_wire_order, wire_order=wire_order)
 
+    def _to_sparse_mat(self, wire_order):
+        """Compute the sparse matrix of the Pauli sentence by efficiently adding the Pauli words
+        that conform it. See pauli_sparse_matrices.md for the technical details."""
+        pauli_words = list(self)  # Ensure consistent ordering
+        n_wires = len(wire_order)
+        matrix_size = 2**n_wires
+        matrix = sparse.csr_matrix((matrix_size, matrix_size), dtype="complex128")
+        op_sparse_idx = _ps_to_sparse_index(pauli_words, wire_order)
+        _, unique_inds, unique_invs = np.unique(
+            op_sparse_idx, axis=0, return_index=True, return_inverse=True
+        )
+        pw_sparse_structures = unique_inds[unique_invs]
+
+        buffer_size = 2 ** max(0, 31 - n_wires - 5)  # Buffer of 1GB memory
+        mat_data = np.empty((matrix_size, buffer_size), dtype=np.complex128)
+        mat_indices = np.empty((matrix_size, buffer_size), dtype=np.int64)
+        i = 0
+        for sparse_structure in unique_inds:
+            idx = np.where(pw_sparse_structures == sparse_structure)[0]
+            mat = self._sum_same_structure_pws([pauli_words[i] for i in idx], wire_order)
+            mat_data[:, i] = mat.data
+            mat_indices[:, i] = mat.indices
+
+            i += 1
+            if i == buffer_size:
+                # Add partial results in batches to control the memory usage
+                matrix += self._sum_different_structure_pws(mat_indices, mat_data)
+                i = 0
```

Comment:
Once we have `SPARSE_MATRIX_BUFFER_SIZE` in option, we should parametrize a test with `SPARSE_MATRIX_BUFFER_SIZE=1` or some small value that hit that part.